app/components/product-destroy/template.hbs

```diff
@@ -0,0 +1,8 @@
+<button data-test-product-destroy-button {{action "launchConfirmDialog"}}>{{yield}}</button>
+{{#if confirmShown}}
```

Comment:
7 lines below look like a separate component - `confirm-action` interface of this general purpose component could look like:

```` hbs
{{confirm-action onConfirm=(action 'delete') warningMsg="This will remove the product, do you wish to proceed?" }}
````

And the action `delete` should be implemented on `product-destroy` component (not in product-thumb)